pg_search/src/api/aggregate.rs

```diff
@@ -277,62 +310,92 @@ pub fn aggregate(
+            let mut nlaunched = process.launched_workers();
+            if pg_sys::parallel_leader_participation {
+                nlaunched += 1;
+            }
 
-        process
-            .state_manager_mut()
-            .object::<State>(0)?
-            .unwrap()
-            .set_launched_workers(nlaunched);
-
-        // leader participation
-        let mut agg_results = Vec::with_capacity(nlaunched);
-        if pg_sys::parallel_leader_participation {
-            let mut worker = ParallelAggregationWorker::new(*process.state_manager());
-            if let Some(result) = worker.execute_aggregate(-1)? {
-                agg_results.push(Ok(result));
+            process
+                .state_manager_mut()
+                .object::<State>(0)?
+                .unwrap()
+                .set_launched_workers(nlaunched);
+
+            // leader participation
+            let mut agg_results = Vec::with_capacity(nlaunched);
+            if pg_sys::parallel_leader_participation {
+                let mut worker = ParallelAggregationWorker::new(*process.state_manager());
+                if let Some(result) = worker.execute_aggregate(-1)? {
+                    agg_results.push(Ok(result));
+                }
             }
-        }
 
-        // wait for workers to finish, collecting their intermediate aggregate results
-        for (_worker_number, message) in process {
-            let worker_results = postcard::from_bytes::<IntermediateAggregationResults>(&message)?;
+            // wait for workers to finish, collecting their intermediate aggregate results
+            for (_worker_number, message) in process {
+                let worker_results =
+                    postcard::from_bytes::<IntermediateAggregationResults>(&message)?;
 
-            agg_results.push(Ok(worker_results));
-        }
+                agg_results.push(Ok(worker_results));
+            }
+
+            // have tantivy finalize the intermediate results from each worker
+            let merged = {
+                let collector = DistributedAggregationCollector::from_aggs(
+                    agg_req.clone(),
+                    AggregationLimitsGuard::new(
+                        Some(memory_limit.try_into()?),
+                        Some(bucket_limit.try_into()?),
+                    ),
+                );
+                collector.merge_fruits(agg_results)?.into_final_result(
+                    agg_req,
+                    AggregationLimitsGuard::new(
+                        Some(memory_limit.try_into()?),
+                        Some(bucket_limit.try_into()?),
+                    ),
+                )?
+            };
 
-        // have tantivy finalize the intermediate results from each worker
-        let merged = {
-            let collector = DistributedAggregationCollector::from_aggs(
+            Ok(JsonB(serde_json::to_value(merged)?))
+        } else {
```

Comment:
Nit: it feels like it would be good for these two branches to be in separate methods (`run_my_stuff_parallel`, `run_my_stuff_local`) so that the symmetry would be more apparent moving forward.